cob_scan_unifier/src/scan_unifier_node.cpp

```diff
@@ -179,14 +201,19 @@ void ScanUnifierNode::messageFilterCallback(const sensor_msgs::LaserScan::ConstP
   current_scans.push_back(scan3);
   current_scans.push_back(scan4);
 
+  sensor_msgs::PointCloud2 unified_point = sensor_msgs::PointCloud2();
   sensor_msgs::LaserScan unified_scan = sensor_msgs::LaserScan();
   if (!unifyLaserScans(current_scans, unified_scan))
   {
     return;
   }
-
+  projector_.transformLaserScanToPointCloud(frame_, unified_scan, unified_point, listener_);
   ROS_DEBUG("Publishing unified scan.");
   topicPub_LaserUnified_.publish(unified_scan);
+  if(config_.pub_point_cloud)
+  {
+  topicPub_PointcloudUnified_.publish(unified_point);
+  }
```

Comment:
there is a lot of duplicated code here. is it possible to move that to "unifyLaserScans" or create a "publish" that handles publishing scan and point cloud